lib/ansible/modules/find.py

```diff
@@ -547,7 +547,7 @@ def handle_walk_errors(e):
     looked = 0
     has_warnings = False
     for npath in params['paths']:
-        npath = os.path.expanduser(os.path.expandvars(npath))
+        npath = os.path.expandvars(npath)
```

Comment:
this change is incorrect, expandvars will not handle `~user/dirs`

Reply:
So the semantics of `type="path"` does not apply to `type="list", elements=" path"` , is that it?

Reply:
@bcoca 
So, just added a test that differs from your statement. Please review.

Reply:
My point is that your change is incorrect in this line as it does not do the same function. Since you defined them as `path` the whole line can be removed as that is doing the exact same thing as the original line you are edditing here.
```python
def check_type_path(value,):
    """Verify the provided value is a string or convert it to a string,
    then return the expanded path
    """
    value = check_type_str(value)
    return os.path.expanduser(os.path.expandvars(value))
```

Reply:
Done! Thanks for pointing it out.